eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Epoch.hs

```diff
@@ -121,13 +123,15 @@ import Data.Maybe (fromMaybe)
 import Data.Maybe.Strict (StrictMaybe (..))
 import Data.Sequence.Strict (StrictSeq (..))
 import qualified Data.Sequence.Strict as Seq
+import Data.Set (Set)
 import qualified Data.Set as Set
 import Data.Void (Void, absurd)
 import Lens.Micro (Lens', (%~), (&), (+~), (.~), (<>~), (^.))
 
 data ConwayEpochEvent era
   = PoolReapEvent (Event (EraRule "POOLREAP" era))
   | SnapEvent (Event (EraRule "SNAP" era))
+  | RatifyEvent (RatifyEnv era) (EnactState era) (Set (GovActionId (EraCrypto era)))
```

Comment:
Let's call it `EnactEvent` instead, because actual ratification will happen at an arbitrary point within the epoch, while act of enactment will happen at known time onthe epoch boundary. Also, if you really need something specific from the `RatifyEnv` we could potentially include it in the event, but I'd suggest we do not include the whole thing in the event, since its contents are quite complicated and will soon change. 
```suggestion
  | EnactEvent (EnactState era) (Set (GovActionId (EraCrypto era)))
```